sixtyfps_runtime/corelib/model.rs

```diff
@@ -205,14 +205,51 @@ impl Model for bool {
     }
 }
 
+/// A shared model
+#[repr(transparent)]
+#[derive(derive_more::Deref, derive_more::DerefMut)]
+pub struct SharedModel<T>(pub Rc<dyn Model<Data = T>>);
+
+impl<T> std::fmt::Debug for SharedModel<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "SharedModel(dyn Model)")
+    }
+}
+
+impl<T> Clone for SharedModel<T> {
+    fn clone(&self) -> Self {
+        Self(self.0.clone())
+    }
+}
+
+impl<T> core::cmp::PartialEq for SharedModel<T> {
+    #[allow(clippy::vtable_address_comparisons)]
+    // We think vtable_address_comparisons are safe here!
+    // Rc::ptr_eq will compare a fat pointer consisting of a `vtable` and a `self` pointer.
+    //
+    // We can get:
+    // * false positives: Two different `Model`s get considered the same, even though they are not.
+    //   For this the `self` pointer need to point to the same place. That can only happen if
+    //   `Self` is `()`, which does not make sense for a `Model`. So we ignore this case.
+    // * false negatives: The same `Model` gets considered to be two different `Model`s.
+    //   In this case the `vtable` pointer must differ, e.g. because two crates created separate
+    //   instances of the vtable for the `Model` trait.
+    //   This will trigger some unnecessary rendering, but is benign otherwise.
+    fn eq(&self, other: &Self) -> bool {
+        Rc::ptr_eq(&self.0, &other.0)
+    }
+}
+
 /// Properties of type array in the .60 language are represented as
 /// an [`Option`] of an [`Rc`] of something implemented the [`Model`] trait
-#[derive(derive_more::Deref, derive_more::DerefMut, derive_more::From, derive_more::Into)]
-pub struct ModelHandle<T>(pub Option<Rc<dyn Model<Data = T>>>);
+#[derive(
+    PartialEq, derive_more::Deref, derive_more::DerefMut, derive_more::From, derive_more::Into,
+)]
+pub struct ModelHandle<T>(pub Option<SharedModel<T>>);
```

Comment:
i wonder if there is a way to remove the inner `pub`

Anyway, this is a public API breakage.

Reply:
True, I added something about this to the CHANGELOG now.